### PR TITLE
Add mesh shader support: Support mesh shader register configuration

### PR DESCRIPTION
**Pull request by amdrexu**
Add mesh shader register configuration. Also, support on-chip GS
calculation for mesh shader since mesh shader is mapped to NGG
fast-launch (could be viewed as a special HW GS).

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 984770d0f7d5190ecb116613b9bc21e678fbc40a
<details><summary>CTS tests (Failed: 0/216059)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35298/73410 (48.1%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 38112/73410 (51.9%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 33017/69239 (47.7%)</li>
<li>Failed: 0/69239 (0.0%)</li>
<li>Not Supported: 36222/69239 (52.3%)</li><li>Warnings: 0/69239 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37553/73410 (51.2%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 35857/73410 (48.8%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/Gfx9ConfigBuilder.cpp [hunk 1]
```diff
@@ -1934,23 +1943,208 @@ template <typename T> void ConfigBuilder::buildPsRegConfig(ShaderStage shaderSta
...
+  SET_REG_FIELD(&config->meshRegs, PA_CL_VTE_CNTL, VPORT_X_OFFSET_ENA, true);
+  SET_REG_FIELD(&config->meshRegs, PA_CL_VTE_CNTL, VPORT_Y_SCALE_ENA, true);
+  SET_REG_FIELD(&config->meshRegs, PA_CL_VTE_CNTL, VPORT_Y_OFFSET_ENA, true);
+  SET_REG_FIELD(&config->meshRegs, PA_CL_VTE_CNTL, VPORT_Z_SCALE_ENA, true);
+  SET_REG_FIELD(&config->meshRegs, PA_CL_VTE_CNTL, VPORT_Z_OFFSET_ENA, true);
+  SET_REG_FIELD(&config->meshRegs, PA_CL_VTE_CNTL, VTX_W0_FMT, true);
+
+  SET_REG_FIELD(&config->meshRegs, PA_SU_VTX_CNTL, PIX_CENTER, 1);
+  SET_REG_FIELD(&config->meshRegs, PA_SU_VTX_CNTL, ROUND_MODE, 2); // Round to even
+  SET_REG_FIELD(&config->meshRegs, PA_SU_VTX_CNTL, QUANT_MODE, 5); // Use 8-bit fractions
```
Comment (nhaehnle) [inline lgc/patch/Gfx9ConfigBuilder.cpp:2073]: Can the PA_* register settings be shared with other geometry stages? (There are a few more below)

Comment (amdrexu) [reply lgc/patch/Gfx9ConfigBuilder.cpp:2073]: Yes, I think so. I will factor out such piece of codes.

Comment (amdrexu) [reply lgc/patch/Gfx9ConfigBuilder.cpp:2073]: > Can the PA_* register settings be shared with other geometry stages? (There are a few more below)

Fixed by https://github.com/GPUOpen-Drivers/llpc/pull/1951. Will rebase this PR after it.

**Comment by qiaojbao** [comment]
14:48:05  /jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/llpc/lgc/patch/Gfx9ConfigBuilder.cpp:1662:15: error: unused variable ‘inOutUsage’ [-Werror=unused-variable]
14:48:05     const auto &inOutUsage = resUsage->inOutUsage;
14:48:05                 ^~~~~~~~~~
14:48:05  /jenkins/workspace/vulkan/san

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4e46815df2514defb04e65f527659414c43dfedb
<details><summary>CTS tests (Failed: 0/289625)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx9</summary><ul>

<li>Passed: 32868/69268 (47.5%)</li>
<li>Failed: 0/69268 (0.0%)</li>
<li>Not Supported: 36400/69268 (52.5%)</li><li>Warnings: 0/69268 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi21</summary><ul>

<li>Passed: 38491/73498 (52.4%)</li>
<li>Failed: 0/73498 (0.0%)</li>
<li>Not Supported: 35007/73498 (47.6%)</li><li>Warnings: 0/73498 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 34348/73361 (46.8%)</li>
<li>Failed: 0/73361 (0.0%)</li>
<li>Not Supported: 39013/73361 (53.2%)</li><li>Warnings: 0/73361 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35388/73498 (48.1%)</li>
<li>Failed: 0/73498 (0.0%)</li>
<li>Not Supported: 38110/73498 (51.9%)</li><li>Warnings: 0/73498 (0.0%)</li>
</ul></details>


</ul></details>

**Review by linqun — APPROVED** [review]

File: lgc/patch/Gfx9ConfigBuilder.cpp (hunk 1 above)
Comment (nhaehnle) [reply lgc/patch/Gfx9ConfigBuilder.cpp]: Thank you for doing this!

**Pull request merged**